pulling in pip internals causes depreciation warning

```
import isort
../../../../virtualenv/python3.7.1/lib/python3.7/site-packages/isort/__init__.py:26: in <module>
    from .isort import SortImports  # noqa: F401
../../../../virtualenv/python3.7.1/lib/python3.7/site-packages/isort/isort.py:41: in <module>
    from .finders import FindersManager
../../../../virtualenv/python3.7.1/lib/python3.7/site-packages/isort/finders.py:24: in <module>
    from pip._internal.download import PipSession
../../../../virtualenv/python3.7.1/lib/python3.7/site-packages/pip/_internal/__init__.py:40: in <module>
    from pip._internal.cli.autocompletion import autocomplete
../../../../virtualenv/python3.7.1/lib/python3.7/site-packages/pip/_internal/cli/autocompletion.py:8: in <module>
    from pip._internal.cli.main_parser import create_main_parser
../../../../virtualenv/python3.7.1/lib/python3.7/site-packages/pip/_internal/cli/main_parser.py:12: in <module>
    from pip._internal.commands import (
../../../../virtualenv/python3.7.1/lib/python3.7/site-packages/pip/_internal/commands/__init__.py:6: in <module>
    from pip._internal.commands.completion import CompletionCommand
../../../../virtualenv/python3.7.1/lib/python3.7/site-packages/pip/_internal/commands/completion.py:6: in <module>
    from pip._internal.cli.base_command import Command
../../../../virtualenv/python3.7.1/lib/python3.7/site-packages/pip/_internal/cli/base_command.py:25: in <module>
    from pip._internal.index import PackageFinder
../../../../virtualenv/python3.7.1/lib/python3.7/site-packages/pip/_internal/index.py:14: in <module>
    from pip._vendor import html5lib, requests, six
../../../../virtualenv/python3.7.1/lib/python3.7/site-packages/pip/_vendor/html5lib/__init__.py:25: in <module>
    from .html5parser import HTMLParser, parse, parseFragment
../../../../virtualenv/python3.7.1/lib/python3.7/site-packages/pip/_vendor/html5lib/html5parser.py:8: in <module>
    from . import _tokenizer
../../../../virtualenv/python3.7.1/lib/python3.7/site-packages/pip/_vendor/html5lib/_tokenizer.py:16: in <module>
    from ._trie import Trie
../../../../virtualenv/python3.7.1/lib/python3.7/site-packages/pip/_vendor/html5lib/_trie/__init__.py:3: in <module>
    from .py import Trie as PyTrie
../../../../virtualenv/python3.7.1/lib/python3.7/site-packages/pip/_vendor/html5lib/_trie/py.py:6: in <module>
    from ._base import Trie as ABCTrie
../../../../virtualenv/python3.7.1/lib/python3.7/site-packages/pip/_vendor/html5lib/_trie/_base.py:3: in <module>
    from collections import Mapping
<frozen importlib._bootstrap>:1032: in _handle_fromlist
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
name = 'Mapping'
    def __getattr__(name):
        # For backwards compatibility, continue to make the collections ABCs
        # through Python 3.6 available through the collections module.
        # Note, no new collections ABCs were added in Python 3.7
        if name in _collections_abc.__all__:
            obj = getattr(_collections_abc, name)
            import warnings
            warnings.warn("Using or importing the ABCs from 'collections' instead "
                          "of from 'collections.abc' is deprecated, "
                          "and in 3.8 it will stop working",
>                         DeprecationWarning, stacklevel=2)
E           DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated, and in 3.8 it will stop working
../../../../virtualenv/python3.7.1/lib/python3.7/collections/__init__.py:52: DeprecationWarning
```

Since pip has internal copy of html5 with the deprecated use, it's very hard to manage this as an isort package user.

--- a/isort/finders.py
+++ b/isort/finders.py
@@ -20,15 +20,9 @@ except ImportError:
     pipreqs = None
 
 try:
-    # pip>=10
-    from pip._internal.download import PipSession
-    from pip._internal.req import parse_requirements
+    from pip_api import parse_requirements
 except ImportError:
-    try:
-        from pip.download import PipSession
-        from pip.req import parse_requirements
-    except ImportError:
-        parse_requirements = None
+    parse_requirements = None
 
 try:
     from requirementslib import Pipfile
@@ -325,8 +319,8 @@ class RequirementsFinder(ReqsBaseFinder):
         result = []
 
         with chdir(os.path.dirname(path)):
-            requirements = parse_requirements(path, session=PipSession())
-            for req in requirements:
+            requirements = parse_requirements(path)
+            for req in requirements.values():
                 if req.name:
                     result.append(req.name)
 

--- a/setup.py
+++ b/setup.py
@@ -24,7 +24,7 @@ setup(name='isort',
       extras_require={
           'pipfile': ['pipreqs', 'requirementslib'],
           'pyproject': ['toml'],
-          'requirements': ['pip', 'pipreqs'],
+          'requirements': ['pip', 'pipreqs', 'pip-api'],
           'xdg_home': ['appdirs>=1.4.0'],
       },
       python_requires=">=3.4",

--- a/test_isort.py
+++ b/test_isort.py
@@ -2578,8 +2578,6 @@ def test_new_lines_are_preserved():
         os.remove(n_newline.name)
 
 
-@pytest.mark.skipif(not finders.RequirementsFinder.enabled, reason='RequirementsFinder not enabled (too old version of pip?)')
-@pytest.mark.skipif(not finders.pipreqs, reason='pipreqs is missing')
 def test_requirements_finder(tmpdir):
     subdir = tmpdir.mkdir('subdir').join("lol.txt")
     subdir.write("flask")
@@ -2656,8 +2654,6 @@ deal = {editable = true, git = "https://github.com/orsinium/deal.git"}
 """
 
 
-@pytest.mark.skipif(not finders.PipfileFinder.enabled, reason='PipfileFinder not enabled (missing requirementslib?)')
-@pytest.mark.skipif(not finders.pipreqs, reason='pipreqs is missing')
 def test_pipfile_finder(tmpdir):
     pipfile = tmpdir.join('Pipfile')
     pipfile.write(PIPFILE)

--- a/test_isort.py
+++ b/test_isort.py
@@ -2597,7 +2597,7 @@ def test_requirements_finder(tmpdir):
         files = list(finder._get_files())
         assert len(files) == 1  # file finding
         assert files[0].endswith('requirements.txt')  # file finding
-        assert list(finder._get_names(str(req_file))) == ['Django', 'deal']  # file parsing
+        assert set(finder._get_names(str(req_file))) == {'Django', 'deal'}  # file parsing
 
         assert finder.find("django") == si.sections.THIRDPARTY  # package in reqs
         assert finder.find("flask") is None  # package not in reqs

